src/Functions/FunctionConstantBase.h

```diff
@@ -42,6 +42,12 @@ class FunctionConstantBase : public IFunction
 
     ColumnPtr executeImpl(const ColumnsWithTypeAndName &, const DataTypePtr & result_type, size_t input_rows_count) const override
     {
+        if (is_distributed)
+        {
+            auto result_column = result_type->createColumn();
```

Comment:
There is `convertToFullColumnIfConst` function.

Reply:
Discussed with @KochetovNicolai , we decided that returning a non-const column here is not the best solution, because we should not rely on column constness.
This fix aim to fix the behavior in new analyzer. And in the current analyzer, we explicitly add `materialize` function. It's not the best approach either, but it's worth doing the same thing in the new analyzer.

Reply:
Ok, I just pointed that createColumnConst(...)->convertToFullColumnIfConst() is often used in the codebase for this purpose.